libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -589,9 +589,11 @@ def _create_chat_result(
         # Otherwise, the next code 'choices.extend(response["choices"])'
         # will throw a "TypeError: 'NoneType' object is not iterable" error
         # to mask the true error. Because 'response["choices"]' is None.
-        if response.get("error"):
-            raise ValueError(response.get("error"))
-
+        if 'error' in response:
+            error_message = response['error']['message']
+            raise ValueError(error_message)
+        if 'choices' not in response or response['choices'] is None:
+            raise ValueError("No choices available in the response, please check the input and model configuration.")
```

Comment:
why is raising an error here better than returning an empty list of Generations?

Reply:
Indeed, your point is well-taken.  But, the absence of 'choices' might not only be due to user errors but also service interruptions. Returning an empty list may potentially lead to errors in downstream processes, e.g. some users might take an empty list as an empty string. I think it's generally better to alert users directly about the problem so they can troubleshoot effectively. This ensures clarity and helps maintain system integrity.

Reply:
if no choices were returned due to a server error wouldn't there be 'error' key in the response (which is already checked)